code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -37,31 +37,50 @@ LINEN BINS
 	if(bedsheet_type == BEDSHEET_DOUBLE)
 		stack_amount *= 2
 		dying_key = DYE_REGISTRY_DOUBLE_BEDSHEET
+	register_item_context()
+
+/obj/item/bedsheet/add_item_context(datum/source, list/context, mob/living/target)
+	context[SCREENTIP_CONTEXT_ALT_LMB] = "Rotate"
+
+	if(isliving(target) && target.body_position == LYING_DOWN)
+		context[SCREENTIP_CONTEXT_LMB] = "Cover"
+
+	return CONTEXTUAL_SCREENTIP_SET
+
+/obj/item/bedsheet/attack(mob/living/target, mob/living/user)
+	if(!user.CanReach(target))
+		return
+	if(user.combat_mode)
+		return ..()
+	if(target.body_position != LYING_DOWN)
+		return
+	if(!user.dropItemToGround(src))
+		return
+
+	forceMove(get_turf(target))
+	balloon_alert(user, "covered")
+	coverup(target)
+	add_fingerprint(user)
 
 /obj/item/bedsheet/attack_self(mob/living/user)
 	if(!user.CanReach(src)) //No telekenetic grabbing.
 		return
+	if(user.body_position != LYING_DOWN)
+		return
 	if(!user.dropItemToGround(src))
 		return
-	if(layer == initial(layer))
-		layer = ABOVE_MOB_LAYER
-		SET_PLANE_IMPLICIT(src, GAME_PLANE_UPPER)
-		to_chat(user, span_notice("You cover yourself with [src]."))
-		pixel_x = 0
-		pixel_y = 0
-	else
-		layer = initial(layer)
-		SET_PLANE_IMPLICIT(src, initial(plane))
-		to_chat(user, span_notice("You smooth [src] out beneath you."))
-	if(user.body_position == LYING_DOWN)    //The player isn't laying down currently
-		dir = user.dir
-	else
-		if(user.dir & WEST)    //The player is rotated to the right, lay the sheet left!
-			dir = WEST
-		else    //The player is rotated to the left, lay the sheet right!
-			dir = EAST
+
+	coverup(user)
 	add_fingerprint(user)
-	return
+
+/obj/item/bedsheet/proc/coverup(mob/living/sleeper)
+	layer = ABOVE_MOB_LAYER
+	SET_PLANE_IMPLICIT(src, GAME_PLANE_UPPER)
```

Comment:
register COMSIG_MOVABLE_MOVED on the sleeper to set the layer back to initial once they're gone?

Reply:
Done with https://github.com/tgstation/tgstation/pull/70576/commits/415291adb27136c4fa936080c022858419e12349